src/javascript/audio.js

```diff
@@ -0,0 +1,64 @@
+import { Howl, Howler } from "howler";
+
+export default class Audio {
+  constructor() {
+    this.audioData = {
+      audioCtx: null,
+      analyser: null,
+      bufferLength: null,
+      dataArray: null,
+    };
+    this.player = null;
+  }
+
+  async loadSong(songSrc, asBackground) {
+    if (this.player) this.player.unload();
+
+    this.player = new Howl({
+      src: [songSrc],
+      loop: asBackground,
+    });
+
+    // ref https://stackoverflow.com/questions/32460123/connect-analyzer-to-howler-sound
+
+    // Create an analyser node in the Howler WebAudio context
+    let analyser = Howler.ctx.createAnalyser();
+
+    // Connect the masterGain -> analyser (disconnecting masterGain -> destination)
+    Howler.masterGain.connect(analyser);
+
+    analyser.fftSize = 256;
+
+    this.audioData.bufferLength = analyser.frequencyBinCount;
+
+    this.audioData.dataArray = new Uint8Array(this.audioData.bufferLength);
+
+    this.audioData.analyser = analyser;
+    this.audioData.audioCtx = Howler.ctx;
+
+    if (asBackground) this.player.play();
+  }
+
+  stop() {
+    if (this.player) this.player.stop();
```

Comment:
We can make this much better IF we use ES2017+ syntax (see the JavaScript nullish coalescing operator `?.`).

We can then do
```javascript
this.player?.stop();
```

Which, semantically, means to invoke `stop()` on `this.player` if and only if `this.player` is not null. I would recommend using this over `this.player` checks!

Reply:
Make sense! That's indeed much better, will change that later